File: gesture/src/main/java/com/omarea/gesture/util/ScreenState.java
package com.omarea.gesture.util;

import android.app.KeyguardManager;
import android.content.Context;
import android.os.Build;
import android.view.Display;
import android.view.WindowManager;

public class ScreenState {
    private Context context;

    public ScreenState(Context context) {
        this.context = context;
    }

    public boolean isScreenLocked() {
        WindowManager windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        Display display = windowManager.getDefaultDisplay();
        if (display.getState() == Display.STATE_OFF) {
            return true;
        }

        KeyguardManager mKeyguardManager = (KeyguardManager) context.getSystemService(Context.KEYGUARD_SERVICE);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            return mKeyguardManager.isDeviceLocked() || mKeyguardManager.isKeyguardLocked();
        } else {
            return mKeyguardManager.inKeyguardRestrictedInputMode() || mKeyguardManager.isDeviceLocked() || mKeyguardManager.isKeyguardLocked();
        }
    }

    public boolean isScreenOn() {
        return !isScreenLocked();
    }
}

File: app/src/main/java/com/omarea/library/basic/ScreenState.kt
package com.omarea.library.basic

import android.app.KeyguardManager
import android.content.Context
import android.os.Build
import android.view.Display
import android.view.WindowManager

class ScreenState(private var context: Context) {
    private val mKeyguardManager = context.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
    private val screenOffStatus = intArrayOf(
            Display.STATE_OFF,
            Display.STATE_DOZE,
            Display.STATE_DOZE_SUSPEND,
            Display.STATE_ON_SUSPEND
    ).toTypedArray()

    fun isScreenLocked(): Boolean {
        val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val display = windowManager.defaultDisplay
        if (screenOffStatus.contains(display.state)) {
            return true
        }

        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            mKeyguardManager.isDeviceLocked || mKeyguardManager.isKeyguardLocked
        } else {
            mKeyguardManager.inKeyguardRestrictedInputMode() || mKeyguardManager.isDeviceLocked || mKeyguardManager.isKeyguardLocked
        }
    }

    fun isScreenOn(): Boolean {
        return !isScreenLocked()
    }
}
